include/multipass/virtual_machine.h

```diff
@@ -90,15 +90,23 @@ class VirtualMachine : private DisabledCopyMove
     using SnapshotVista = std::vector<std::shared_ptr<const Snapshot>>; // using vista to avoid confusion with C++ views
     virtual SnapshotVista view_snapshots() const noexcept = 0;
     virtual int get_num_snapshots() const noexcept = 0;
+
     virtual std::shared_ptr<const Snapshot> get_snapshot(const std::string& name) const = 0;
+    virtual std::shared_ptr<const Snapshot> get_snapshot(int index) const = 0;
     virtual std::shared_ptr<Snapshot> get_snapshot(const std::string& name) = 0;
+    virtual std::shared_ptr<Snapshot> get_snapshot(int index) = 0;
+
     virtual std::shared_ptr<const Snapshot> take_snapshot(const VMSpecs& specs,
                                                           const std::string& name,
```

Comment:
Here, missed this one. However, looking around some more, we use `name` for `delete_snapshot()` and `restore_snapshot()`. So, I don't expect you to change all of them.

Reply:
Right, all of these methods use "snapshot" as the (grammatical) object, so I think that makes it relatively obvious that the parameter refers to that same entity. I think this is a pattern we use in other places in the code too. In the case of get/delete/restore, I think it is very obvious (get/delete/restore the snapshot with this name). 

In the case of `take_snapshot`, the name could conceivably be part of the contents that are being snapshot, like the specs are. But we're already operating on the VM object, which knows its name, so we wouldn't receive it as a parameter. I remember thinking about this when I wrote that code and concluding that there wouldn't be a real ambiguity. That said, I was not in the shoes of the reader. Were you actually confused by this, or is it just a matter of consistency?

Honestly, I think the real problem is that specs need to be received in a parameter. One would expect them to be part of the VM too. But that is just the fact of the matter until we dedicate some significant time to that refactor. So OK, after organizing my thoughts a little I think there is special reason to rename the parameter in the case of `take_snapshot`, that doesn't apply to the other `*_snapshot` methods. I will do that.

Reply:
Sounds good to me. To me, it more just a matter of consistency.

As for the specs, yes I remember seeing a comment elsewhere in the codebase before snapshots saying that they should be part of the vm.

Reply:
Yeah, I added that comment :wink:

Reply:
Alright done